crates/ergot/src/net_stack/inner.rs

```diff
@@ -356,7 +356,10 @@ where
                 continue;
             }
             if skt_ref.attrs.kind != hdr.kind {
-                return Err(NetStackSendError::WrongPortKind);
+                return Err(NetStackSendError::WrongPortKind {
+                    expected: skt_ref.attrs.kind,
```

Comment:
maybe we want to call these something like "socket_kind" and "packet_kind", to make it more clear which is which?